### DISCUSSION TITLE
Example of using Qualifier.byAnnotation()

### DISCUSSION THREAD

**Discussion by ratcashdev** [micronaut-projects/micronaut-core#7192] [Q&A] [answered]
Can anyone please give an example, how to use Qualifier.byAnnotation(<annotation>) to retrieve a list of classes implementing a specific annotation (in a dependency jar)? The annotation in question is not a micronaut @Qualifier.
Or an example of how to retrieve a collection of all the jakarta @Singleton annotated classes in the current project and dependency jars?
Thank you.

**Comment by jeffscottbrown** [comment]
> Can anyone please give an example, how to use Qualifier.byAnnotation() to retrieve a list of classes implementing a specific annotation

There are examples around the core code base. Ex:

[inject/src/main/java/io/micronaut/inject/BeanDefinition.java#L432](https://github.com/micronaut-projects/micronaut-core/blob/7e3226394942006d006c2dbc8c875b2172bc0351/inject/src/main/java/io/micronaut/inject/BeanDefinition.java#L432)

[inject/src/main/java/io/micronaut/inject/BeanDefinition.java#L435](https://github.com/micronaut-projects/micronaut-core/blob/7e3226394942006d006c2dbc8c875b2172bc0351/inject/src/main/java/io/micronaut/inject/BeanDefinition.java#L435)

[inject/src/main/java/io/micronaut/inject/BeanDefinition.java#L446](https://github.com/micronaut-projects/micronaut-core/blob/7e3226394942006d006c2dbc8c875b2172bc0351/inject/src/main/java/io/micronaut/inject/BeanDefinition.java#L446)

**Comment by ratcashdev** [comment, marked as answer]
Thanks Jeff.
As I reference, I had to to do the following:
```java
BeanContext ctx = BeanContext.run();
Annotation singletonAnnotation = () -> Singleton.class;
    Collection<Object> beansOfType =
        beanContext.getBeansOfType(Object.class, Qualifiers.byAnnotation(singletonAnnotation));
System.out.println("beansOfType = " + beansOfType);
```
